### PR TITLE
Make handling scheduleResult more readable

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: Kante Yin <kerthcet@gmail.com>

#### What type of PR is this?
/kind cleanup
/sig scheduling

#### What this PR does / why we need it:

This is a prefactor for https://github.com/kubernetes/kubernetes/issues/113854

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/schedule_one.go [hunk 1]
```diff
@@ -130,42 +130,41 @@ func (sched *Scheduler) schedulingCycle(
...
 		// SchedulePod() may have failed because the pod would not fit on any host, so we try to
 		// preempt, with the expectation that the next time the pod is tried for scheduling it
 		// will fit due to the preemption. It is also possible that a different pod will schedule
 		// into the resources that were preempted, but this is harmless.
-		var (
-			nominatingInfo *framework.NominatingInfo
-			status         *framework.Status
-		)
 		if fitError, ok := err.(*framework.FitError); ok {
+			var nominatingInfo *framework.NominatingInfo
```
Comment (ahg-g) [inline pkg/scheduler/schedule_one.go:144]: to further reduce indentation, I suggest the following:

```
fitError, ok := err.(*framework.FitError)
if !ok {
  klog.ErrorS(err, "Error selecting node for pod", "pod", klog.KObj(pod))
  return ScheduleResult{nominatingInfo: clearNominatedNode}, podInfo, framework.AsStatus(err)
}

// the rest
```

File: pkg/scheduler/schedule_one.go [hunk 2]
```diff
@@ -130,42 +130,41 @@ func (sched *Scheduler) schedulingCycle(
...
+
 			if !fwk.HasPostFilterPlugins() {
 				klog.V(3).InfoS("No PostFilter plugins are registered, so no preemption will be performed")
+				return ScheduleResult{}, podInfo, framework.NewStatus(framework.Unschedulable).WithError(err)
+			}
+			// Run PostFilter plugins to try to make the pod schedulable in a future scheduling cycle.
```
Comment (ahg-g) [inline pkg/scheduler/schedule_one.go:150]:
```suggestion
			// Run PostFilter plugins to try to attempt to make the pod schedulable in a future scheduling cycle.
```

File: pkg/scheduler/schedule_one.go (hunk 1 above)
Comment (kerthcet) [reply pkg/scheduler/schedule_one.go:144]: Yeah, that makes sense, updated.

File: pkg/scheduler/schedule_one.go [hunk 3]
```diff
@@ -130,42 +130,43 @@ func (sched *Scheduler) schedulingCycle(
...
+		result, status := fwk.RunPostFilterPlugins(ctx, state, pod, fitError.Diagnosis.NodeToStatusMap)
+		if status.Code() == framework.Error {
+			klog.ErrorS(nil, "Status after running PostFilter plugins for pod", "pod", klog.KObj(pod), "status", status)
 		} else {
-			klog.ErrorS(err, "Error selecting node for pod", "pod", klog.KObj(pod))
-			nominatingInfo = clearNominatedNode
-			status = framework.AsStatus(err)
+			msg := status.Message()
+			fitError.Diagnosis.PostFilterMsg = msg
+			klog.V(5).InfoS("Status after running PostFilter plugins for pod", "pod", klog.KObj(pod), "status", msg)
```
Comment (ahg-g) [inline pkg/scheduler/schedule_one.go:163]: when https://github.com/kubernetes/kubernetes/pull/114456 merges, we will get the error in the reasons and will be returned as part of status.Message(), so we can change this if/else to 

```
msg := status.Message()
fitError.Diagnosis.PostFilterMsg = msg
if status.Code() == framework.Error {
  klog.ErrorS(nil, "After running PostFilter plugins for pod", "pod", klog.KObj(pod), "status", msg)
} else {
  klog.V(5).InfoS("After running PostFilter plugins for pod", "pod", klog.KObj(pod), "status", msg)
}
```

**Comment by ahg-g** [comment]
please don't amend

File: pkg/scheduler/schedule_one.go (hunk 3 above)
Comment (kerthcet) [reply pkg/scheduler/schedule_one.go:163]: Always record the PostFilter messages whatever the status, make sense, updated.

**Review by ahg-g — COMMENTED** [review]
thanks, please squash

**Comment by kerthcet** [comment]
> thanks, please squash

Done.

**Pull request merged**